besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogSubCommand.java

```diff
@@ -124,12 +129,52 @@ public void run() {
       final Path dataDirectoryPath =
           Paths.get(
               TrieLogSubCommand.parentCommand.parentCommand.dataDir().toAbsolutePath().toString());
+
+      long estimatedSaving = getEstimatedSaving();
+
       final TrieLogHelper trieLogHelper = new TrieLogHelper();
-      trieLogHelper.prune(
-          context.config(),
-          context.rootWorldStateStorage(),
-          context.blockchain(),
-          dataDirectoryPath);
+      boolean success =
+          trieLogHelper.prune(
+              context.config(),
+              context.rootWorldStateStorage(),
+              context.blockchain(),
+              dataDirectoryPath);
+
+      if (success) {
+        spec.commandLine()
+            .getOut()
+            .printf(
+                "Prune ran successfully. We estimate you freed up %s! \uD83D\uDE80\n",
+                RocksDbUsageHelper.formatOutputSize(estimatedSaving));
+      }
+    }
+
+    private long getEstimatedSaving() {
+      final String dbPath =
+          TrieLogSubCommand.parentCommand
+              .parentCommand
+              .dataDir()
+              .toString()
+              .concat("/")
+              .concat(DATABASE_PATH);
+
+      AtomicLong estimatedSaving = new AtomicLong(0L);
+      RocksDbUsageHelper.forEachColumnFamily(
+          dbPath,
+          (rocksdb, cfHandle) -> {
+            try {
+              if (Arrays.equals(cfHandle.getName(), TRIE_LOG_STORAGE.getId())) {
+                estimatedSaving.set(
+                    Long.parseLong(
+                        rocksdb.getProperty(cfHandle, "rocksdb.estimate-live-data-size")));
```

Comment:
This isn't the savings. It's only the estimated savings if a small number of trielogs is kept. Think could be misleading for some people.

If we are going to print the estimated savings then we should compare before and after data size of the trie log column family.

Reply:
I didn't really want to slow the subcommand down further for an edge case. Think this is an acceptable trade-off because it's just an estimate anyway. Would you block the PR on it?

Reply:
Also I don't think reevaluating after will work within the same subcommand without manually triggering the compaction

Reply:
Spiking this out here: https://github.com/hyperledger/besu/pull/6492

Results from local test...

```
Trie logs size before: 578 KiB
Trie logs size after: 0 B
Prune ran successfully. We estimate you freed up 578 KiB! 🚀
```

Reply:
Results from a (not very old and already pruned) mainnet node:
```
2024-01-30 05:51:43.982+00:00 | main | INFO  | RocksDBKeyValueStorageFactory | Existing database detected at /data/besu. Version 2. Compacting database...
2024-01-30 05:51:45.286+00:00 | main | INFO  | TrieLogSubCommand | Calculating trie logs size before...
2024-01-30 05:52:38.295+00:00 | main | INFO  | TrieLogSubCommand | Trie logs size before: 30 MiB
...
2024-01-30 05:52:38.819+00:00 | main | INFO  | TrieLogSubCommand | Finished pruning - recalculating size...
2024-01-30 05:52:39.271+00:00 | main | INFO  | TrieLogSubCommand | Trie logs size after: 0 B
Prune ran successfully. We estimate you freed up 30 MiB! 🚀
```

Compaction was fast, "before" estimate nearly took 1 minute, "after" estimate didn't add any data.

Will try this on an older node next.

Reply:
Tested on a larger node...

```
2024-01-31 03:27:06.481+00:00 | main | INFO  | TrieLogSubCommand | Finished pruning - recalculating size...
2024-01-31 03:27:07.034+00:00 | main | INFO  | TrieLogSubCommand | Trie logs size after: 0 B
Prune ran successfully. We estimate you freed up 8 GiB! 🚀
```

So yeh, I think we'll need to compact to see any "size after"

Reply:
Better results when the retained trie logs value is higher actually...
```
$ sudo /opt/besu/current/bin/besu --data-storage-format=BONSAI --data-path=/data/besu --bonsai-historical-block-limit=20000 storage x-trie-log prune

...
2024-01-31 04:32:04.021+00:00 | main | INFO  | RocksDBKeyValueStorageFactory | Existing database detected at /data/besu. Version 2. Compacting database...
...
2024-01-31 04:32:05.229+00:00 | main | INFO  | TrieLogSubCommand | Calculating trie logs size before...
2024-01-31 04:32:05.683+00:00 | main | INFO  | TrieLogSubCommand | Trie logs size before: 10 GiB
2024-01-31 04:32:17.623+00:00 | main | INFO  | TrieLogHelper | Saving trie logs to retain in file /data/besu/database/trieLogsToRetain-1 (batch 1)...
2024-01-31 04:32:17.625+00:00 | main | INFO  | TrieLogHelper | Obtaining trielogs from db, this may take a few minutes...
2024-01-31 04:32:39.579+00:00 | main | INFO  | TrieLogHelper | Clear trie logs...
2024-01-31 04:32:43.786+00:00 | main | INFO  | TrieLogHelper | Restoring trie logs retained from batch 1...
...
2024-01-31 04:33:46.212+00:00 | main | INFO  | TrieLogHelper | Deleting files...
2024-01-31 04:33:46.294+00:00 | main | INFO  | TrieLogSubCommand | Finished pruning - recalculating size...
2024-01-31 04:33:57.324+00:00 | main | INFO  | TrieLogSubCommand | Trie logs size after: 546 MiB
Prune ran successfully. We estimate you freed up 9 GiB! 🚀

Reply:
Given the results, I think this is worth including this PR, thanks for the feedback 😄